contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -357,51 +359,52 @@ pub fn execute_burn(
         return Err(ContractError::Unauthorized {});
     }
 
-    // make sure is expired if the stage is not paused
-    let is_paused = STAGE_PAUSED.load(deps.storage, stage)?;
-    if !is_paused {
-        let expiration = STAGE_EXPIRATION.load(deps.storage, stage)?;
-        if !expiration.is_expired(&env.block) {
-            return Err(ContractError::StageNotExpired { stage, expiration });
+    // make sure all the stages are either paused or expired
+    let latest_stage = LATEST_STAGE.load(deps.storage)?;
```

Comment:
`execute_burn` should be applied for a single stage. Best to add `burn_all` instead of modifying here.